### DISCUSSION TITLE
New asynio dialect for mysql

### DISCUSSION THREAD

**Discussion by synackSA** [sqlalchemy/sqlalchemy#6893] [Ideas]
Hi there

We currently use SQLAlchemy in our projects and we're in the process of switching our python grpc apps to use asyncio. I noticed that SQLAlchemy support async, but for mysql there is only a single dialect, "aiomysql". I tried implementing this and it seems that it's currently in a broken state. The Github repo for aiomysql (https://github.com/aio-libs/aiomysql) as been touch in over a year and any posts to issues, discussions, or their google groups mailing list goes unanswered.

Are there plans to implement a new/different dialect for SQLAlchemy any time soon to support async requests for MySQL?

**Comment by CaselIT** [comment]
Hi,

I think aiomysql was implemented because is was the most popular for mysql. Do you have an alternative dialect you wanted support?

**Comment by synackSA** [comment]
After searching high and low, this is pretty much the only other driver that I have found that supports async/await and has been recently updated: https://github.com/long2ice/asyncmy

**Reply by CaselIT** [reply]
It does not seem to be that used. It would probably be best if they add support for sqlalchemy by writing a dialect. It was already requested here https://github.com/long2ice/asyncmy/issues/6

Sadly there does not seem to be any "quality"/popular async driver for anything other than pg

**Comment by zzzeek** [comment]
agree the "aio-libs" project is alarming in how prominent they are compared with the apparent state of non-maintenance.

I think the "real" story here is that async shops just use PostgreSQL, because PG has native "async" support at the server level.   

asyncmy looks promising but it's only 6 months old.

**Comment by synodriver** [comment]
Sure ```asyncmy``` need to be tested by time, but currently it's the only drivers which support asyncio，especially ```aiomysql``` appears to be unmaintained.

**Comment by skwzrd** [comment]
Hello,

I am trying to decide if I should be using `aiomysql` or `asyncmy` as a DB API for SQLAlchemy.

I noticed the DB API documentation at https://docs.sqlalchemy.org/en/20/dialects/mysql.html#aiomysql states `aiomysql` is unmaintained since September 2021, however, [new versions](https://github.com/aio-libs/aiomysql/releases/tag/v0.2.0) have been release since then. It also states that `aiomysql` uses a very outdated version of `pymysql`, but [it uses](https://github.com/aio-libs/aiomysql/blob/master/requirements-dev.txt) `pymysql==1.0.3`, which was [released March of this year](https://pypi.org/project/pymysql/#history).

`aiomysql` seems to have a larger user base, so I am partial to using it rather than `asyncmy`.

**Reply by CaselIT** [reply]
Well then the documentation should be updated.
Outside postgresql, all other dbs don't have "good" async drivers thought

**Reply by CaselIT** [reply]
created https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4792 to readd the dialect to the ci and update the docs. let's see how it goes

**Reply by zzzeek** [reply]
disagree.  My issue https://github.com/aio-libs/aiomysql/issues/543 which prevents SQLAlchemy from working correctly is still opened, still **unanswered**, for **three years**.    

Please have aiomysql maintainers respond to my issue if this project is again maintained.

**Reply by CaselIT** [reply]
the test suite seems to have only one fail on an unrelated test though

**Reply by zzzeek** [reply]
Yeah it's probably fixed now.

The main thing is when we have some showstopper here, other project has to have someone to respond to us.  Otherwise I really don't have the ability to hang on to unresponsive projects

**Reply by CaselIT** [reply]
shall we just add it to the ci for the time being and change the note we have now to a different one?
